spa/src/views/app/appViewModel.ts

```diff
@@ -1,29 +1,31 @@
 import {ApiClient} from '../../api/apiClient';
 import {Configuration} from '../../configuration';
 import {OAuthClient} from '../../oauth/oauthClient';
-import {StorageHelper} from '../../utilities/storageHelper';
 
 export class AppViewModel {
 
     public configuration: Configuration | null;
     public oauthClient: OAuthClient | null;
     public apiClient: ApiClient | null;
-    public storage: StorageHelper | null;
+    public initialized: boolean
 
     public constructor() {
         this.configuration = null;
         this.oauthClient = null;
         this.apiClient = null;
-        this.storage = null;
+        this.initialized = false;
     }
 
-    public async initialize(storage: StorageHelper): Promise<void> {
+    public async initialize(): Promise<void> {
 
-        const response = await fetch('config.json');
-        this.configuration = await response.json();
+        if (!this.initialized) {
 
-        this.oauthClient = new OAuthClient(this.configuration!.oauth);
-        this.apiClient = new ApiClient(this.configuration!.businessApiBaseUrl, this.oauthClient);
-        this.storage = storage;
+            const response = await fetch('config.json');
```

Comment:
Apparently, you can use `"resolveJsonModule": true` in `tsconfig` and then json files can be imported in TypeScript: 

```typescript
import config from './config.json'
this.configuration = config
```

See: https://www.typescriptlang.org/tsconfig#resolveJsonModule

Reply:
The code is executing in the browser though, so requires an HTTP request to download it from the web host where the config file lives.